tests/test_cpu.py

```diff
@@ -80,9 +81,108 @@ def test_intel_rapl(self):
         self.assertDictEqual(expected_cpu_details, rapl.get_cpu_details())
 
 
+
 class TestTDP(unittest.TestCase):
-    def test_parse_cpu_model(self):
-        model_raw = "Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz"
-        model_parsed = parse_cpu_model(model_raw)
-        model_parsed_expected = "Intel Core i7-8850H"
-        self.assertEqual(model_parsed, model_parsed_expected)
+    def test_get_cpu_power_from_registry(self):
+        tdp = TDP()
+        model = "Intel Core i7-8850H"
+        self.assertEqual(tdp._get_cpu_power_from_registry(model), 45)
+        model = "AMD Ryzen Threadripper 1950X"
+        self.assertEqual(tdp._get_cpu_power_from_registry(model), 180)
+        model = "AMD Ryzen Threadripper 1950X 16-Core Processor"
+        self.assertEqual(tdp._get_cpu_power_from_registry(model), 180)
+
+
+    def test_get_matching_cpu(self):
```

Comment:
Could we add test cases on obvious non-matches ? fuzzy non-matches ?
To see where is the limit of the CPU parsing.

Reply:
I've added some fuzzy non-matches. Do you think we should find more of them?